src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs

```diff
@@ -0,0 +1,214 @@
+                return apiRevision.IsApproved ? Ok(reviewUrl) : StatusCode(statusCode: StatusCodes.Status201Created, reviewUrl);
+            }
+            // Return internal server error for any unknown error
+            return StatusCode(statusCode: StatusCodes.Status500InternalServerError);
+        }
+
+        private async Task<APIRevisionListItemModel> CreateAutomaticRevisionAsync(CodeFile codeFile, string label, string originalName, MemoryStream memoryStream, bool compareAllRevisions = false)
+        {
+            var createNewRevision = true;
+            var review = await _reviewManager.GetReviewAsync(packageName: codeFile.PackageName, language: codeFile.Language, isClosed: null);
+            var apiRevision = default(APIRevisionListItemModel);
+            var renderedCodeFile = new RenderedCodeFile(codeFile);
+            var apiRevisions = (await _apiRevisionsManager.GetAPIRevisionsAsync(review.Id)).OrderByDescending(r => r.CreatedOn).ToList();
+
+            if (review != null)
+            {
+                if (apiRevisions.Any())
+                {
+                    // Delete pending apiRevisions if it is not in approved state before adding new revision
+                    // This is to keep only one pending revision since last approval or from initial review revision
+                    var automaticRevisions = new Queue<APIRevisionListItemModel>(apiRevisions.Where(r => r.APIRevisionType == APIRevisionType.Automatic));
+                    var latestAutomaticAPIRevision = automaticRevisions.Peek();
+                    var comments = await _commentsManager.GetCommentsAsync(review.Id);
+
+                    while (
+                        automaticRevisions.Any() &&
+                        !latestAutomaticAPIRevision.IsApproved &&
+                        !await _apiRevisionsManager.IsAPIRevisionTheSame(latestAutomaticAPIRevision, renderedCodeFile) &&
+                        !comments.Any(c => latestAutomaticAPIRevision.Id == c.RevisionId))
+                    {
+                        await _apiRevisionsManager.SoftDeleteAPIRevisionAsync(user: User, apiRevision: latestAutomaticAPIRevision);
+                        latestAutomaticAPIRevision = automaticRevisions.Dequeue();
+                    }
+
+                    // We should compare against only latest revision when calling this API from scheduled CI runs
+                    // But any manual pipeline run at release time should compare against all approved revisions to ensure hotfix release doesn't have API change
+                    // If review surface doesn't match with any approved revisions then we will create new revision if it doesn't match pending latest revision
+
+                    if (compareAllRevisions)
+                    {
+                        foreach (var approvedAPIRevision in apiRevisions.Where(r => r.IsApproved))
```

Comment:
Are you saying this should check against automatic revisions?

Reply:
Yes, Otherwise we won't create an automatic revision when new API changes merged to main branch when that change is already present under a PR or manual revision. right?